BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb

```diff
@@ -368,12 +368,25 @@ class CbrainTask::<%= name %> < <%= descriptor['cbrain:inherits-from-class'] ||
           next
         end
 
-        self.addlog("Saving result file #{path}")
-        name = File.basename(path)
-        
+        # Get name and filetype
+        self.addlog("Attempting to save result file #{path}")
+        name = File.basename(path)         
         userfile_class = Userfile.suggested_file_type(name)
-        userfile_class ||= ( File.directory?(path) ? FileCollection : SingleFile )
+        userfile_class ||= ( File.directory?(path) ? FileCollection : SingleFile ) 
+       
+        # If the file already exists, add a tag (task id + ns time) to the name
+        if userfile_exists(userfile_class, :name => name)
+          tag = '-' + (self.id.to_s || 'unk') + '-' + Time.now.nsec.to_s
+          newName = name + tag if (userfile_class == FileCollection) && name.rindex(".").nil?
+          ext = File.extname( name ) 
+          newName ||= name.gsub( /#{ext}$/, "#{tag + ext}" )
```

Comment:
You're right, checking the file type is not really necessary I think. Originally I was going to only append directly to directory objects... but then I realized many tools output directories with names like blah.anat.

Incidentally, something special in the string substitution  must be going on:

```
2.2.4 :031 > "foo".gsub(/''$/,"-a")
 => "foo" 
2.2.4 :032 > "foo".gsub(/#{ File.extname('foo') }$/,"-aa")
 => "foo-aa"
```

which is why it worked when I tested that case.